### DISCUSSION TITLE
FIle Search Widget

### DISCUSSION THREAD

**Discussion by sowjanyagunupuru** [eclipse-theia/theia#14251] [Q&A] [answered]
I have added a widget for file search with basic functionality and ui styles as shown in the attached image 

But it's not handling the input states as expected i.e search term or file

Here I need suggestion , that the filteration must be done among all the existed files.
How can I give that files content to the file search widget 

And how can I make  this widget to handle the input states ime searchterm

![IMG_20241004_214716](https://github.com/user-attachments/assets/83a7572c-9ca5-450d-99c9-880c867f7c5e)

**Comment by msujew** [comment]
Hey @sowjanyagunupuru,

I'm unsure what you want. Can you explain a bit what the search bar should search? Are you expecting it to open something similar to VS Code?

![grafik](https://github.com/user-attachments/assets/8a7537a4-dcd6-4f85-b3a7-6d6449d8cafc)

**Reply by sowjanyagunupuru** [reply]
That search bar should find the files

Like in vscode
We use that to find the respective file by giving that file name as input 

Most matched file names with the input are shown 

I want that functionality for my search bar

**Reply by sowjanyagunupuru** [reply]
Yes as shown in the image shared by u is exactly my requirement

**Reply by msujew** [reply, marked as answer]
I see. You can use the `QuickInputService#open` method by injecting an instance of `QuickInputService` to achieve that. That should automatically open this exact popup.

**Reply by sowjanyagunupuru** [reply]
Yeah it's working perfect as expected
Thank you for your support 😊

**Reply by sowjanyagunupuru** [reply]
Hello @msujew ,

Can u kindly help me adding an extension for icon such as FlatIcon

So that the icons which are shown on the screen will be replaced with the icons of the included extension 

Icons such as shown below in the image must be replaced with the icons of added extension 
But how can we include manage that.?

Is there any method to include the extension and use the icons of that extension instead of existed 

Such that all the icons in the ui screen will be the icons of added extension i.e FlatIcon 

![IMG_20241004_215329](https://github.com/user-attachments/assets/ffcf486c-9580-4ceb-b3e6-0a6d6fd2183c)

**Reply by msujew** [reply]
The CSS for views contributed by VS Code extensions is dynamically generated here:

https://github.com/eclipse-theia/theia/blob/ac0c246fbcd439c259b4778e41199708d4354899/packages/plugin-ext/src/main/browser/plugin-shared-style.ts#L124-L143

You might be able to override this service in a way so that you can replace the paths of the icons with a custom one. I don't think there's a good way to generically replace those icons with the icon of the extension, given that the extension icon is often times not suited to be displayed in the side bar.

**Reply by sowjanyagunupuru** [reply]
I found your suggestion to use QuickInputService#open for file searches extremely helpful in fulfilling my requirement and saving time.

Now, I have another request: I’d like to implement the navigation functionality (left/right arrows) that already exists in the toolbar for my icons. Can you guide me on how to utilize that functionality in my top panel as my icons are in toppanel?

![IMG_20241009_162005](https://github.com/user-attachments/assets/58a87940-d26a-461e-8bf0-5cb2134fb589)

**Reply by msujew** [reply]
> Can you guide me on how to utilize that functionality in my top panel as my icons are in toppanel?

You should be able to run the respective commands (see toolbar) by injecting the `CommandRegistry` into your service.
